packages/dd-trace/src/proxy.js

```diff
@@ -74,4 +76,35 @@ class Tracer extends NoopProxy {
   }
 }
 
+function maybeStartServerlessMiniAgent () {
```

Comment:
Can this be moved out of the proxy since it's serverless specific?

Reply:
Also, how is this tested? Would we need an additional test with a fake backend instead of a fake agent?

Reply:
Yep! that makes sense, I'll move it out of proxy.

The tracer sends traces to the mini agent via HTTP, which listens on the default port the tracer expects (8126). We will definitely want to add integration tests with a fake backend to the mini-agent